### PR TITLE
feat(argo-cd): Use tini as the ENTRYPOINT

### PR DESCRIPTION
**Pull request by yu-croco**
ref: https://github.com/argoproj/argo-helm/pull/1989#issuecomment-1531239945

Signed-off-by: yu-croco <yu.croco@gmail.com>

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.7.0
 kubeVersion: ">=1.22.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 5.30.1
+version: 5.31.0
```
Comment (yu-croco) [inline charts/argo-cd/Chart.yaml:6]: In order to follow upstream (https://github.com/argoproj/argo-cd/commit/c2e0026de16c2622ddf4c95d0311c321ebd51f3a) `.Values.global.entrypoint.useImplicit` is not used anymore.
So I incremented minor version.

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml [hunk 2]
```diff
@@ -284,7 +280,7 @@ spec:
       {{- end }}
       initContainers:
       - command:
-        - cp
+        - /bin/cp
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml:283]: Here is still `command` .
https://github.com/argoproj/argo-cd/blob/v2.7.0/manifests/base/repo-server/argocd-repo-server-deployment.yaml#L198-L201

image

File: charts/argo-cd/README.md [hunk 3]
```diff
@@ -105,6 +105,11 @@ For full list of changes please check ArtifactHub [changelog].
 
 Highlighted versions provide information about additional steps that should be performed by user when upgrading to newer version.
 
+### 5.31.0
+The manifests are now using [`tini` as entrypoint][tinit], instead of `entrypoint.sh`. Until Argo CD v2.8, `entrypoint.sh` is retained for upgrade compatibility.
+This means that the deployment manifests have to be updated after upgrading to Argo CD v2.7, and before upgrading to Argo CD v2.8 later.
+In case the manifests are updated before moving to Argo CD v2.8, the containers will not be able to start.
```
Comment (yu-croco) [inline charts/argo-cd/README.md:111]: I referred https://github.com/argoproj/argo-cd/blob/v2.7.0/docs/operator-manual/upgrading/2.6-2.7.md#tini-as-entrypoint .

File: charts/argo-cd/README.md [hunk 4]
```diff
@@ -105,6 +105,11 @@ For full list of changes please check ArtifactHub [changelog].
 
 Highlighted versions provide information about additional steps that should be performed by user when upgrading to newer version.
 
+### 5.31.0
+The manifests are now using [`tini` as entrypoint][tinit], instead of `entrypoint.sh`. Until Argo CD v2.8, `entrypoint.sh` is retained for upgrade compatibility.
```
Comment (mbevc1) [inline charts/argo-cd/README.md:109]: `tini` instead of `tinit`? :thinking:

Comment (yu-croco) [reply charts/argo-cd/README.md:109]: typo 😇

Comment (yu-croco) [reply charts/argo-cd/README.md:109]: fixed at https://github.com/argoproj/argo-helm/pull/1991/commits/3412931f41066af20bb611a4fa0c15e8ab916402 🙋

File: charts/argo-cd/README.md [hunk 5]
```diff
@@ -1184,3 +1187,4 @@ Autogenerated from chart metadata using [helm-docs](https://github.com/norwoodj/
 [TopologySpreadConstraints]: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
 [values.yaml]: values.yaml
 [v2.2 to 2.3 upgrade instructions]: https://github.com/argoproj/argo-cd/blob/v2.3.0/docs/operator-manual/upgrading/2.2-2.3.md
+[tinit]: https://github.com/argoproj/argo-cd/pull/12707
```
Comment (mbevc1) [inline charts/argo-cd/README.md:1190]: Another typo? :wink:

Comment (yu-croco) [reply charts/argo-cd/README.md:1190]: ah ~~~ `./scripts/helm-docs.sh` 😇

**Review by pdrastil — APPROVED** [review]

**Pull request merged**